public/app/features/explore/RichHistory/RichHistoryCard.test.tsx

```diff
@@ -56,93 +79,140 @@ const starredQueryWithComment: RichHistoryQuery<MockQuery> = {
-    expect(wrapper.find({ 'aria-label': 'Query text' }).at(2).text()).toEqual('{"query":"query3"}');
-  });
-  it('should render data source icon', () => {
-    const wrapper = setup();
-    expect(wrapper.find({ 'aria-label': 'Data source icon' })).toHaveLength(1);
+  it('should render all queries', async () => {
+    setup();
+    const queries = await screen.findAllByLabelText('Query text');
+    expect(queries).toHaveLength(3);
+    expect(queries[0]).toHaveTextContent('query1');
+    expect(queries[1]).toHaveTextContent('query2');
+    expect(queries[2]).toHaveTextContent('query3');
   });
-  it('should render data source name', () => {
-    const wrapper = setup();
-    expect(wrapper.find({ 'aria-label': 'Data source name' }).text()).toEqual('Test datasource');
+  it('should render data source icon and name', async () => {
+    setup();
+    const datasourceIcon = await screen.findAllByLabelText('Data source icon');
+    const datasourceName = await screen.findAllByLabelText('Data source name');
+    expect(datasourceIcon).toHaveLength(1);
+    expect(datasourceName).toHaveLength(1);
   });
-  it('should render "Data source does not exist anymore" if removed data source', () => {
-    const wrapper = setup({ isRemoved: true });
-    expect(wrapper.find({ 'aria-label': 'Data source name' }).text()).toEqual('Data source does not exist anymore');
+  it('should render "Data source does not exist anymore" if removed data source', async () => {
+    setup({ isRemoved: true });
+    const datasourceName = await screen.findByLabelText('Data source name');
+    expect(datasourceName).toHaveTextContent('Data source does not exist anymore');
   });
 
   describe('commenting', () => {
-    it('should render comment, if comment present', () => {
-      const wrapper = setup({ query: starredQueryWithComment });
-      expect(wrapper.find({ 'aria-label': 'Query comment' })).toHaveLength(1);
-      expect(wrapper.find({ 'aria-label': 'Query comment' }).text()).toEqual('test comment');
+    it('should render comment, if comment present', async () => {
+      setup({ query: starredQueryWithComment });
+      const queryComment = await screen.findAllByLabelText('Query comment');
+      expect(queryComment).toHaveLength(1);
+      expect(queryComment[0]).toHaveTextContent('test comment');
     });
-    it('should have title "Edit comment" at comment icon, if comment present', () => {
-      const wrapper = setup({ query: starredQueryWithComment });
-      expect(wrapper.find({ title: 'Edit comment' })).toHaveLength(1);
-      expect(wrapper.find({ title: 'Add comment' })).toHaveLength(0);
+    it('should have title "Edit comment" at comment icon, if comment present', async () => {
+      setup({ query: starredQueryWithComment });
+      const editComment = await screen.findByTitle('Edit comment');
+      const addComment = await screen.queryByTitle('Add comment');
+      expect(editComment).toBeVisible();
+      expect(addComment).toBeNull();
```

Comment:
```suggestion
      const editComment = await screen.findByTitle('Edit comment');
      const addComment = screen.queryByTitle('Add comment');
      expect(editComment).toBeInTheDocument();
      expect(addComment).not.toBeInTheDocument();
```

`queryBy*` is not async so there's no need for `await` there. Also while there's nothing wrong with `toBeVisible` and `toBeNull` checks, I think that asserting that the element is in the DOM is a bit more clear when reading the tests (unless in case of `toBeVisible` we do need to assert the CSS properties of an element).